docs/security-policy.md

```diff
@@ -0,0 +1,112 @@
+# NSX Operator Security Policy CRD
+
+## Summary
+
+Standard K8s NetworkPolicies secures traffic between Pods, a cluster administrator
+may want to have the ability to configure consistent network policy through a common
+API for Namespaces which apply to both Pods and VMs in K8s cluster. This document
+describes SecurityPolicy CRD to support the security control of workloads within
+the cluster.
+SecurityPolicy CRD is used to apply NSX-T based security policy to VMs and Pods in
+K8s cluster. VMs should be defined as workload resources in K8s cluster, both VMs
+and Pods are attached to NSX-T segments, each workload has segment port on NSX-T.
+Labels on VMs and Pods are showed on Tags of their segment ports.
+In the first step, nsx-operator leverages Tags added by NSX Container Plugin(NCP)
+on NSX-T segment ports, nsx-operator needs to work together with NCP on Supervisor
+cluster of vSphere with Kubernetes. 
+nsx-operator will reconcile SecurityPolicy CRs, calls NSX-T API to create
+NSX-T Distributed Firewall rules, then updates CR status with realized state.
+
+## CRD Design
+
+SecurityPolicy is Namespaced scope CRD, admins could apply CR within a namespace to
+configure security of workloads which are attached to NSX-T networking.
+Compared with standard K8s NetworkPolicies, SecurityPolicy CRD adds `VMSelector` to
+select VMs, it also introduces some rule syntaxes including Drop and Reject actions,
+priority, rule level selectors(`appliedTo`). 
+
+An example of SecurityPolicy CR:
+
+```yaml
+apiVersion: nsx.vmware.com/v1alpha1
+kind: SecurityPolicy
+metadata:
+  name: db-isolation
+  namespace: prod-ns
+spec:
+  priority: 1
+  appliedTo:
+    - vmSelector:
+        matchLabels:
+          role: db
+  rules:
+    - direction: in
+      action: allow
+      sources:
+        - namespaceSelector:
+            matchLabels:
+              role: control
+        - podSelector:
+            matchLabels:
+              role: frontend
+      ports:
+        - protocol: TCP
+          port: 8000
+    - direction: out
+      action: allow
+      destinations:
+        - podSelector:
+            matchLabels:
+              role: dns
+      ports:
+        - protocol: UDP
+          port: 53
+    - direction: in
+      action: drop
+    - direction: out
+      action: drop
+status:
+  conditions:
+    - type: "Ready"
+      status: True
+      reason: "SuccessfulRealized"
+```
+
+The example CR defines security rules for VMs which match the label "role: db"
+in Namespace prod-ns. The first rule allows traffic from Namespaces match label
+"role: control" or Pods match label "role: frontend" in Namespace prod-ns to
```

Comment:
"Namespaces match label" -> "Namespaces matching label" and "Pods match label" -> "Pods matching label" ?